packages/patternfly-4/react-charts/src/components/ChartUtils/chart-domain.ts

```diff
@@ -0,0 +1,130 @@
+import * as React from "react";
+import { DataGetterPropType } from 'victory';
+import { Data } from "victory-core";
+
+interface DomainInterface {
+  data?: any;
+  maxDomain?: any;
+  minDomain?: any;
+  x?: DataGetterPropType;
+  y?: DataGetterPropType;
+}
+
+interface SourcesInterface {
+  maxDomain?: any;
+  minDomain?: any;
+  sources?: {
+    component?: React.ReactElement<any>;
+    data?: any;
+  }[];
+}
+
+export type ChartDomain = {x: [number, number], y: [number, number]};
+
+// Returns the min and max domain for given data
+export const getDomain = ({
+  data,
+  maxDomain,
+  minDomain,
+  x,
+  y
+}: DomainInterface): ChartDomain => {
+  // x-domain
+  let xLow = 0;
+  let xHigh = 0;
+  let xLowSearch = true;
+  let xHighSearch = true;
+  if (minDomain !== undefined && minDomain.x !== undefined) {
+    xLow = minDomain.x;
+    xLowSearch = false;
+  } else if (minDomain !== undefined && minDomain.y === undefined) {
+    xLow = minDomain;
+    xLowSearch = false;
+  }
+  if (maxDomain !== undefined && maxDomain.x !== undefined) {
+    xHigh = maxDomain.x;
+    xHighSearch = false;
+  } else if (maxDomain !== undefined && maxDomain.y === undefined) {
+    xHigh = maxDomain;
+    xHighSearch = false;
+  }
+
+  // y-domain
+  let yLow = 0;
+  let yHigh = 0;
+  let yLowSearch = true;
+  let yHighSearch = true;
+  if (minDomain !== undefined && minDomain.y !== undefined) {
+    yLow = minDomain.y;
+    yLowSearch = false;
+  } else if (minDomain !== undefined && minDomain.x === undefined) {
+    yLow = minDomain;
+    yLowSearch = false;
+  }
+  if (maxDomain !== undefined && maxDomain.y !== undefined) {
+    yHigh = maxDomain.y;
+    yHighSearch = false;
+  } else if (maxDomain !== undefined && maxDomain.x === undefined) {
+    yHigh = maxDomain;
+    yHighSearch = false;
+  }
+
+  // Search data for max / min range
+  const datum = Data.formatData(data, {x, y}, ['x', 'y']);
+  datum.forEach((dataPoint: any) => {
+    if (xLowSearch) {
+      if (dataPoint._x < xLow) {
+        xLow = dataPoint._x;
+      }
+    }
+    if (xHighSearch) {
+      if (dataPoint._x > xHigh) {
+        xHigh = dataPoint._x;
+      }
+    }
+    if (yLowSearch) {
+      if (dataPoint._y < yLow) {
+        yLow = dataPoint._y;
+      }
+    }
+    if (yHighSearch) {
+      if (dataPoint._y > yHigh) {
+        yHigh = dataPoint._y;
+      }
+    }
+  });
+  return {x: [xLow, xHigh], y: [yLow, yHigh]};
+};
+
+export const getDomains = ({
```

Comment:
Where is `getDomains` used? If it's a helper function for the consumer, we can include it in the docs.

Reply:
It's used by the ChartBullet component to figure out the min/max domain based on the given data points. The utility / helper functions are intended for use by chart components -- don't really want/need to expose that in the top level doc.

Reply:
Ah -- I missed that it's used in the component when I used Ctrl+F on Github. Thanks.